From 29e15694b6b3e0cfc1c797cfcc0b834f79176aa1 Mon Sep 17 00:00:00 2001
From: SeongJae Park <sj@kernel.org>
Date: Thu, 11 Apr 2024 16:04:44 -0700
Subject: [PATCH] ==== commit cleanup ====

mm/damon: implement DAMON parameters online commit function

DAMON context struct (damon_ctx) contains user requests (parameters),
internal status, and monitoring/DAMOS results/statistics.  DAMON API
users are encouraged to manually manipulate the struct, for flexible
usages.  That works well for normal use cases.  However, it has turned
out that it is not that simple for at least online parameters udpate.
It is easy to forget keeping and properly updating internal status and
operation results.  Also, the manual manipulation is implemented
multiple times on DAMON API users.  As a result, we have multiple source
of bugs for same problem.  Actually we found and fixed a few bugs from
online parameter updating of DAMON API users.

Implement a function for online DAMON parameters update in core layer,
and replace DAMON API users' manual manipulation code for the use case.
The core layer function could still have bugs, but this change reduces
the source of bugs for the problem to one place.

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 damon_meta_changes/RYnTeJnM | 0
 1 file changed, 0 insertions(+), 0 deletions(-)
 create mode 100644 damon_meta_changes/RYnTeJnM

diff --git a/damon_meta_changes/RYnTeJnM b/damon_meta_changes/RYnTeJnM
new file mode 100644
index 000000000000..e69de29bb2d1
-- 
2.39.2

